src/test/java/org/example/controller/EmployeeControllerTest.java

```diff
@@ -0,0 +1,145 @@
+
+        // Create a test department for employee assignment
+        DepartmentModel testDepartment = new DepartmentModel();
+        testDepartment.setName("Engineering");
+        testDepartment = departmentRepository.save(testDepartment);
+
+        testDepartmentId = testDepartment.getId();
+    }
+
+    @Test
+    public void testCreateEmployee() throws Exception {
+        mockMvc.perform(post("/employees")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("""
+                        {
+                            "name": "John Doe",
+                            "gender": "Male",
+                            "dateOfBirth": "1990-01-01",
+                            "graduationDate": "2012-06-15",
+                            "departmentId": 1,
+                            "managerId": null,
+                            "teamId": 1,
+                            "expertise": ["Java", "Spring Boot"],
+                            "grossSalary": 75000.0
+                        }
+                    """))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value("John Doe"))
+                .andExpect(jsonPath("$.gender").value("Male"))
+                .andExpect(jsonPath("$.dateOfBirth").value("1990-01-01"))
+                .andExpect(jsonPath("$.graduationDate").value("2012-06-15"))
+                .andExpect(jsonPath("$.departmentId").value(1))
+                .andExpect(jsonPath("$.managerId").value(nullValue()))  // Ensure managerId is null
+                .andExpect(jsonPath("$.teamId").value(1))
+                .andExpect(jsonPath("$.expertise[0]").value("Java"))  // Check the expertise array
+                .andExpect(jsonPath("$.expertise[1]").value("Spring Boot"))
+                .andExpect(jsonPath("$.grossSalary").value(75000.0));
+    }
+
+    @Test
+    public void testCreateEmployeeWithInvalidData() throws Exception {
+        // Missing name field
+        mockMvc.perform(post("/employees")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(String.format("""
+                        {
+                            "gender": "Male",
+                            "dateOfBirth": "1990-01-01",
+                            "graduationDate": "2012-06-15",
+                            "expertise": "Software Development",
+                            "grossSalary": 60000.0,
+                            "departmentId": %d
+                        }
+                        """, testDepartmentId)))
+                .andExpect(status().isBadRequest());
+
+        // Invalid graduation date
+        mockMvc.perform(post("/employees")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(String.format("""
+                        {
+                            "name": "Jane Doe",
+                            "gender": "Female",
```

Comment:
same , plus we don't test multiple things in the same tes , so please seperate this is a new one

Reply:
also where are the remaining test cases